net/respondd/src/respondd.c

```diff
@@ -292,20 +288,15 @@ static void add_provider(const char *name, const struct respondd_provider_info *
 	*pos = pentry;
 }
 
-static void load_providers(void) {
+static void load_providers(const char *path) {
 	update_time();
 
-	/* Maximum number of request types, might be made configurable in the future */
-	if (!hcreate_r(32, &htab)) {
-		perror("hcreate_r");
-		exit(EXIT_FAILURE);
-	}
+	if (!chdir(path))
```

Comment:
You still need to chdir back at some point, otherwise relative paths not given as first argument will be interpreted incorrectly.